### ISSUE TITLE
Custom Converters don't work together with multiple identical Objects (fail with ClassCastException)

### ISSUE THREAD

**Issue by ralphhm** [johncarl81/parceler#222]
If I use a custom converter like the following
```
@ParcelClass(value = Date.class, annotation = @Parcel(converter = DateConverter.class))
public class DateConverter implements ParcelConverter<Date> {
    @Override
    public void toParcel(Date input, android.os.Parcel parcel) {
        parcel.writeLong(input.getTime());
    }

    @Override
    public Date fromParcel(android.os.Parcel parcel) {
        return new Date(parcel.readLong());
    }
}
```
in a parcable class like
```
@Parcel
public class ParcelerTestModel {

    final Date firstDate;
    final Date secondDate;
    final String someString;

    @ParcelConstructor
    public ParcelerTestModel(Date firstDate, Date secondDate, String someString) {
        this.firstDate = firstDate;
        this.secondDate = secondDate;
        this.someString = someString;
    }
}
```
and a test class that provides to identical (Date-)object references to the model and parcels/unparcels the object
```
@RunWith(RobolectricGradleTestRunner.class)
public class ParcelerTestModelTest {

    @Test
    public void parcelUnparcel() {
        Date date = new Date();
        ParcelerTestModel model = new ParcelerTestModel(date, date, "Test");
        Parcelable parcelable = Parcels.wrap(model);
        Parcel parcel = Parcel.obtain();
        parcelable.writeToParcel(parcel, 0);
        parcel.setDataPosition(0);
        Parcels.unwrap(((Parcelable.Creator<? extends Parcelable>) ParcelerTestModel$$Parcelable.CREATOR).createFromParcel(parcel));
    }

}
```
it leads to a class cast exception
```
java.lang.ClassCastException: java.lang.String cannot be cast to java.lang.Long

	at org.robolectric.shadows.ShadowParcel$ByteBuffer.readLong(ShadowParcel.java:252)
	at org.robolectric.shadows.ShadowParcel.nativeReadLong(ShadowParcel.java:117)
	at android.os.Parcel.nativeReadLong(Parcel.java)
	at android.os.Parcel.readLong(Parcel.java:1575)
	at DateConverter.fromParcel(DateConverter.java:18)
```
The problem seems to be that in the generated Date$$Parcelable.read() method
```
    public static java.util.Date read(android.os.Parcel parcel$$3, IdentityCollection identityMap$$1) {
        int identity$$1 = parcel$$3 .readInt();
        if (identityMap$$1 .containsKey(identity$$1)) {
            if (identityMap$$1 .isReserved(identity$$1)) {
                throw new ParcelerRuntimeException("An instance loop was detected whild building Parcelable and deseralization cannot continue.  This error is most likely due to using @ParcelConstructor or @ParcelFactory.");
            }
            return identityMap$$1 .get(identity$$1);
        } else {
            return new DateConverter().fromParcel(parcel$$3);
        }
    }
```
that is missing a line to put the unparceled (Date-)object in the else case into the identity map so that the generated code looks similar to this
```
    public static java.util.Date read(android.os.Parcel parcel$$3, IdentityCollection identityMap$$1) {
        int identity$$1 = parcel$$3 .readInt();
        if (identityMap$$1 .containsKey(identity$$1)) {
            if (identityMap$$1 .isReserved(identity$$1)) {
                throw new ParcelerRuntimeException("An instance loop was detected whild building Parcelable and deseralization cannot continue.  This error is most likely due to using @ParcelConstructor or @ParcelFactory.");
            }
            return identityMap$$1 .get(identity$$1);
        } else {
            java.util.Date date = new com.brands4friends.b4f.util.DateConverter().fromParcel(parcel$$3);
            identityMap$$1.put(identity$$1, date);
            return date;
        }
    }
```

**Comment by johncarl81** [comment]
Excellent write-up @ralphhm, ill check it out immediately.

**Comment by johncarl81** [comment]
This should fix the issue: #223

**Comment by johncarl81** [comment]
Would you like to review the change @ralphhm and confirm it fixes your issue?
